app/ide-desktop/lib/content/src/index.ts

```diff
@@ -119,104 +119,132 @@ function displayDeprecatedVersionDialog() {
-                     * where it will be called only once. */
-                    let appInstanceRan = false
-                    const onAuthenticated = () => {
-                        if (
-                            !contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value
-                        ) {
-                            hideAuth()
-                            if (!appInstanceRan) {
-                                appInstanceRan = true
-                                void appInstance.run()
-                            }
-                        }
-                    }
-                    authentication.run({
-                        logger,
-                        platform,
-                        projectManager: projectManager.ProjectManager.default(),
-                        showDashboard:
-                            contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value,
-                        onAuthenticated,
-                    })
-                } else {
-                    void appInstance.run()
-                }
-                const email = contentConfig.OPTIONS.groups.authentication.options.email.value
-                // The default value is `""`, so a truthiness check is most appropriate here.
-                if (email) {
-                    logger.log(`User identified as '${email}'.`)
-                }
-            }
+    currentAppInstance = new app.App({
+        config,
+        configOptions: contentConfig.OPTIONS,
+        packageInfo: {
+            version: BUILD_INFO.version,
+            engineVersion: BUILD_INFO.engineVersion,
+        },
+    })
+
+    if (!currentAppInstance.initialized) {
+        console.error('Failed to initialize the application.')
+    } else {
+        if (contentConfig.OPTIONS.options.dataCollection.value) {
+            // TODO: Add remote-logging here.
+        }
+        if (!(await checkMinSupportedVersion(contentConfig.OPTIONS))) {
+            displayDeprecatedVersionDialog()
         } else {
-            console.error('Failed to initialize the application.')
+            const email = contentConfig.OPTIONS.groups.authentication.options.email.value
+            // The default value is `""`, so a truthiness check is most appropriate here.
+            if (email) {
+                logger.log(`User identified as '${email}'.`)
+            }
+            void currentAppInstance.run()
         }
     }
 }
 
-const API = new Main()
-
-// @ts-expect-error `globalConfig.windowAppScopeName` is not known at typecheck time.
-// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-window[GLOBAL_CONFIG.windowAppScopeName] = API
+if (
+    (contentConfig.OPTIONS.options.authentication.value ||
+        contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value) &&
+    contentConfig.OPTIONS.groups.startup.options.entry.value ===
+        contentConfig.OPTIONS.groups.startup.options.entry.default
+) {
```

Comment:
But previously with `main` implementation we were re-running this function multiple times to open new projects. The `main` function was responsible for:
1. Opening user authentication module then accordingly to the options run either old IDE or new Dashbord
2. Run old IDE without authentication
3. Run IDE with debug scenes
4. We would need to complicate the conditionals to allows us to reinitialise the enso when we wanned to openProject (using the same main)

This violates the single responsibility principle

Now we have a global conditional that initialise the IDE/dashboard (or e.g. debug scenes), separate `runEnso` method used for opening projects and `tryStopEnso` which conditionally `drop`s old instance

Reply:
yeah idk, `index.ts` *is* the entrypoint so there needs to be *some* global code - and i'm not too sure wrapping all top level code in a `main()` function and calling that is any better

as for global variables:
- this is a ESM module, the variables don't ever leak out of this file, let alone into `window`
- we can just wrap everything in a scope `{ all code here }` to be extra safe though

if the main function follows our code style better, then it *is* easy enough to switch
```ts
function main() {
  // all top level code here
}

main()
```

Reply:
agree with refactoring to variables though. i think a better solution in the long run would be maybe to make the structure of `contentConfig` simpler, but that's definitely out of scope for this PR

Reply:
@PabloBuchu I do not understand your comment. If a function violates single responsibility principle, it can be refactored to multiple functions. There is never a need to extract code to top-level of the module and create bunch of global variables.

@somebody1234 The thing is, that if we are not using global variables and global code, it is way easier to refactor this code in the future. In particular, if all such variables are stored as class fields, and all functions are this class methods, then refactoring such code is as simple as copy-pasting it to another file. If things are defined globally, it is way harder to refactor it to other files where variable names could alias themselves after refactoring. Another topic here is that structuring everything in classes with well designed responsibilities makes the code easier to read and understand. So yes, let's not create global, module level logic.